scripts/globals/chocobo_digging.lua

```diff
@@ -803,204 +837,318 @@ local digInfo =
 
 local function updatePlayerDigCount(player, increment)
     if increment == 0 then
-        player:setVolatileCharVar('[DIG]DigCount', 0)
+        player:setCharVar('[DIG]DigCount', 0)
     else
-        player:setVolatileCharVar('[DIG]DigCount', player:getCharVar('[DIG]DigCount') + increment)
+        player:setCharVar('[DIG]DigCount', player:getCharVar('[DIG]DigCount') + increment)
     end
 
-    player:setVolatileCharVar('[DIG]LastDigTime', os.time())
-end
-
---[[ Not Implemented
-local function updateZoneDigCount(zoneId, increment)
-    local serverVar = '[DIG]ZONE' .. zoneId .. '_ITEMS'
-
-    -- 0 means we wanna wipe (probably only gonna happen onGameDay or something)
-    if increment == 0 then
-        SetServerVariable(serverVar, 0)
-    else
-        SetServerVariable(serverVar, GetServerVariable(serverVar) + increment)
-    end
+    player:setCharVar('[DIG]LastDigTime', os.time())
 end
-]]--
 
 local function canDig(player)
-    local digCount = player:getCharVar('[DIG]DigCount')
-    local lastDigTime = player:getCharVar('[DIG]LastDigTime')
-    local zoneItemsDug = GetServerVariable('[DIG]ZONE'..player:getZoneID()..'_ITEMS')
-    local zoneInTime = player:getLocalVar('ZoneInTime')
-    local currentTime = os.time()
-    local skillRank = player:getSkillRank(xi.skill.DIG)
-
+    local digCount                 = player:getCharVar('[DIG]DigCount')
+    local lastDigTime              = player:getCharVar('[DIG]LastDigTime')
+    local lastDigX                 = player:getCharVar('[DIG]LastDigX')
+    local lastDigY                 = player:getCharVar('[DIG]LastDigY')
+    local lastDigZ                 = player:getCharVar('[DIG]LastDigZ')
+    local posTable                 = player:getPos()
+    local currX           = math.floor(posTable.x)
+    local currY           = math.floor(posTable.y)
+    local currZ           = math.floor(posTable.z)
+    local distanceSquared = (lastDigX - currX) * (lastDigX - currX) + (lastDigY - currY) * (lastDigY - currY) + (lastDigZ - currZ) * (lastDigZ - currZ)
+    local zoneInTime               = player:getLocalVar('ZoneInTime')
+    local currentTime     = os.time()
+    local skillRank                = player:getSkillRank(xi.skill.DIG)
+     -- personal dig caps
+    local digCap          = DIG_FATIGUE + (skillRank * 10)
     -- base delay -5 for each rank
-    local digDelay = 16 - (skillRank * 5)
-    local areaDigDelay = 60 - (skillRank * 5)
+    local digDelay        = 16 - (skillRank * 5)
```

Comment:
Sometimes this didn't have a minimum so we should add a min, (animation minimum is 4s?)

Reply:
This has been changed, but clamped below

Reply:
You need to unrequest changes